### PR TITLE
[Wallet] Empty wallet + some bugfix

### LINKED ISSUE #3679: iOS: ‘NaN’ text is shown with ‘Subtotal’ or ‘Celo Gold’ on ‘Buy GLD’ screen when user has 0 dollar & 0 gold in Wallet
**Frequency:** 100%
**iOS App version:** iOS Testflight build : v1.7.1 (28)
**Repro on iOS device:** iPhone 11 Pro (13.3)
**Testing account:** 7864900848/Backup key

**Prerequisite:** User should have 0 dollar & 0 gold

**Repro Steps:** 
1) Launch the app and Login with test account
2) Navigate to ‘Gold’ tab and Make sure only ‘Buy’ button is shown 
3) Tap on ‘Buy’ button and Observe 

**Current Behavior:** ‘NaN’ text is shown with ‘Subtotal’ or ‘Celo Gold’ on ‘Buy GLD’ screen 

**Expected Behavior:** Respective gold amount should be shown on ‘Bug GLD’ screen 

**Investigation:** Issue does not reproduce in Android 

**Attachment:** [iOS_NaN is shown at subtotal](https://drive.google.com/file/d/1GhD1AVk7-C3xZFybw6c_0Mf9MnBD-eSd/view?usp=sharing) (Observe the issue between 0:10 to 0:13 secs)

### PR DESCRIPTION
**Pull request by i1skn**
### Description

* Remove `ImportWalletEmpty` screen and implement a Dialog instead https://www.figma.com/file/zt7aTQ5wuXycIwxq5oAmF9/Wallet--Refresh?node-id=6490%3A3591
* Fix redundant PIN ask screen during Account Key set-up.
* Some small bugfix

### Tested

iOS emulator

### Related issues

- Related to https://github.com/celo-org/celo-monorepo/issues/3645
- Fix #3679

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/account/AccountKeyEducation.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function AccountKeyEducation(props: Props) {
     if (props.route.params?.nextScreen) {
       navigate(props.route.params?.nextScreen)
     } else {
-      navigateProtected(Screens.BackupPhrase)
```
Comment (jmrossy) [inline packages/mobile/src/account/AccountKeyEducation.tsx:19]: Should we remove navigateProtected now?

File: packages/mobile/src/import/ImportWallet.tsx [hunk 2]
```diff
@@ -97,6 +101,8 @@ export class ImportWallet extends React.Component<Props, State> {
   }
 
   onPressRestore = () => {
+    const { route, navigation } = this.props
+    const useEmptyWallet = !!route.params?.showZeroBalanceModal
```
Comment (jmrossy) [inline packages/mobile/src/import/ImportWallet.tsx:105]: Why is this in the nav props instead of the redux store? Just for convenience?

**Review by jmrossy — COMMENTED** [review]
LGTM just some questions below

File: packages/mobile/src/account/AccountKeyEducation.tsx (hunk 1 above)
Comment (i1skn) [reply packages/mobile/src/account/AccountKeyEducation.tsx:19]: Wow, man. I've just realised it was the only place it was used... Will remove it, yeah!

File: packages/mobile/src/import/ImportWallet.tsx (hunk 2 above)
Comment (i1skn) [reply packages/mobile/src/import/ImportWallet.tsx:105]: Well, it is only specific to this very screen and does not affect any business logic outside of it. Nav prop felt right for me.

**Review by jmrossy — APPROVED** [review]

**Pull request merged**